### PR TITLE
Package reason-standard.0.1.0~alpha.1

### PR DESCRIPTION
**Pull request by Dean177**
### `reason-standard.0.1.0-alpha.1`
A portable standard library enhancement for Reason and OCaml.
A portable standard library enhancement for Reason and OCaml.

---
* Homepage: https://github.com/dean177/reason-standard
* Source repo: git+https://github.com/dean177/reason-standard.git
* Bug tracker: https://github.com/dean177/reason-standard/issues

---
:camel: Pull-request generated by opam-publish v2.0.2

### CODE REVIEW

**Comment by mseri** [comment]
Fails with ```
#=== ERROR while compiling reason-standard.0.1.0-alpha.1 ======================#
1767# context     2.0.5 | linux/x86_64 | ocaml-variants.4.10.0+rc2 | file:///home/travis/build/ocaml/opam-repository
1768# path        ~/.opam/ocaml-variants.4.10.0+rc2/.opam-switch/build/reason-standard.0.1.0-alpha.1
1769# command     ~/.opam/opam-init/hooks/sandbox.sh build dune build -p reason-standard -j 1 @install @runtest
1770# exit-code   1
1771# env-file    ~/.opam/log/reason-standard-21866-4c2636.env
1772# output-file ~/.opam/log/reason-standard-21866-4c2636.out
1773### output ###
1774# Error: Program refmt not found in the tree or in PATH
1775#  (context: default)
1776# Hint: try: opam install reason
```
During tests

**Comment by mseri** [comment]
It’s almost there. In ocaml 4.06 and 4.07 it is failing with 
```
#=== ERROR while compiling reason-standard.0.1.0-alpha.1 ======================#
# context              2.0.6 | linux/x86_64 | ocaml-base-compiler.4.07.1 | pinned(https://github.com/Dean177/reason-standard/archive/0.1.0-alpha.1.tar.gz)
# path                 ~/.opam/4.07/.opam-switch/build/reason-standard.0.1.0-alpha.1
# command              ~/.opam/4.07/bin/dune build -p reason-standard -j 72 @install
# exit-code            1
# env-file             ~/.opam/log/reason-standard-23-ef9acb.env
# output-file          ~/.opam/log/reason-standard-23-ef9acb.out
### output ###
#     ocamldep native/src/.standard.objs/Standard.ml.d (exit 2)
# (cd _build/default && /home/opam/.opam/4.07/bin/ocamldep.opt -modules -impl native/src/Standard.ml) > _build/default/native/src/.standard.objs/Standard.ml.d
# File "native/src/Standard.ml", line 24, characters 7-10:
# Error: Syntax error: operator expected.
#       ocamlc native/src/.standard.objs/byte/standard__Core.{cmi,cmti} (exit 2)
# (cd _build/default && /home/opam/.opam/4.07/bin/ocamlc.opt -w -40 -g -bin-annot -I native/src/.standard.objs/byte -I /home/opam/.opam/4.07/lib/base -I /home/opam/.opam/4.07/lib/base/base_internalhash_types -I /home/opam/.opam/4.07/lib/base/caml -I /home/opam/.opam/4.07/lib/base/shadow_stdlib -I /home/opam/.opam/4.07/lib/sexplib0 -I /home/opam/.opam/4.07/lib/zarith -no-alias-deps -open Standard__ -o native/src/.standard.objs/byte/standard__Core.cmi -c -intf native/src/Core.mli)
# File "native/src/Core.mli", line 2831, characters 39-47:
# Error: Unbound module Result
```
Which could be solved with stdcompat for example, maybe also stdlib-shims could be enough.

With older ocaml versions it fails with
```
#=== ERROR while compiling reason-standard.0.1.0-alpha.1 ======================#
# context              2.0.6 | linux/x86_64 | ocaml-base-compiler.4.05.0 | pinned(https://github.com/Dean177/reason-standard/archive/0.1.0-alpha.1.tar.gz)
# path                 ~/.opam/4.05/.opam-switch/build/reason-standard.0.1.0-alpha.1
# command              ~/.opam/4.05/bin/dune build -p reason-standard -j 72 @install
# exit-code            1
# env-file             ~/.opam/log/reason-standard-23-daf7c7.env
# output-file          ~/.opam/log/reason-standard-23-daf7c7.out
### output ###
#     ocamldep native/src/.standard.objs/Standard.ml.d (exit 2)
# (cd _build/default && /home/opam/.opam/4.05/bin/ocamldep.opt -modules -impl native/src/Standard.ml) > _build/default/native/src/.standard.objs/Standard.ml.d
# File "native/src/Standard.ml", line 24, characters 7-10:
# Error: Syntax error: operator expected.
#     ocamldep native/src/.standard.objs/Core.mli.d (exit 2)
# (cd _build/default && /home/opam/.opam/4.05/bin/ocamldep.opt -modules -intf native/src/Core.mli) > _build/default/native/src/.standard.objs/Core.mli.d
# File "native/src/Core.mli", line 2140, characters 7-8:
# Error: Syntax error: operator expected.
#     ocamldep native/src/.standard.objs/Core.ml.d (exit 2)
# (cd _build/default && /home/opam/.opam/4.05/bin/ocamldep.opt -modules -impl native/src/Core.ml) > _build/default/native/src/.standard.objs/Core.ml.d
# File "native/src/Core.ml", line 813, characters 7-8:
# Error: Syntax error: operator expected.
```

My suggestion is to simply require ocaml 4.08 or newer

File: packages/reason-standard/reason-standard.0.1.0-alpha.1/opam [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+license: "MIT"
+homepage: "https://github.com/dean177/reason-standard"
+bug-reports: "https://github.com/dean177/reason-standard/issues"
+depends: [
+  "alcotest" {with-test}
+  "base" {>= "0.13.0"}
+  "zarith" {>= "1.9.1"}
+  "dune" {>= "2.4"}
+  "reason" {>= "3.4.0"}
+]
```
Comment (mseri) [inline packages/reason-standard/reason-standard.0.1.0-alpha.1/opam]:
```suggestion
  “ocaml” {>= "4.08.0"}
]
```

Comment (mseri) [reply packages/reason-standard/reason-standard.0.1.0-alpha.1/opam]: After all there cannot be an opam switch without an ocaml compiler, so it is not adding extra dependencies

Comment (Dean177) [reply packages/reason-standard/reason-standard.0.1.0-alpha.1/opam]: Thanks for your help @mseri

File: packages/reason-standard/reason-standard.0.1.0-alpha.1/opam [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+license: "MIT"
+homepage: "https://github.com/dean177/reason-standard"
+bug-reports: "https://github.com/dean177/reason-standard/issues"
+depends: [
+  "alcotest" {with-test}
+  "base" {>= "0.13.0"}
+  "zarith" {>= "1.9.1"}
+  "dune" {>= "2.4"}
+  "reason" {>= "3.4.0"}
+  “ocaml” {>= "4.08.0"}
```
Comment (mseri) [inline packages/reason-standard/reason-standard.0.1.0-alpha.1/opam]:
```suggestion
  "ocaml" {>= "4.08.0"}
```

That’s why I should never do reviews from the tablet...

**Comment by kit-ty-kate** [comment]
The version number of this package doesn't seem correct as `0.1.0-alpha.1` > `0.1.0`.
For `0.1.0` to be chosen above this version, the version number should be: `0.1.0~alpha.1` (with a `~` instead of a `-`). See version numbering in the documentation: https://opam.ocaml.org/doc/Manual.html#Package-Formulas

**Comment by Dean177** [comment]
@kit-ty-kate does this mean I need to close this PR and create another?

**Comment by mseri** [comment]
You can just git mv the file to the correct folder and push the changes to this PR

**Comment by mseri** [comment]
Thanks!

**Comment by mseri** [comment]
Feel free to announce it on the discuss.ocaml.org forum. We have a community category and a announcement tag for this purpose

**Pull request merged**